matplotcheck/raster.py

```diff
@@ -56,112 +56,174 @@ def assert_colorbar_range(self, crange):
-    ):
+        # TODO: return boolean instead of a none value - true if it matches,
+        #  false if it does not match
+
+        #
+        # for label_option in expected_labels:
+        #     if label_option == label:
+        #         return label_option
+
+        label_check = {}
+
+        # Iteratively test each label found in the plot legend to see if it is
+        # in the list of expected labels
+        for i, label in enumerate(labels):
+            test_output = any(
+                label in expected_label
+                for expected_label in expected_labels[i]
+            )
+            label_check[label] = test_output
+
+        return label_check
+
+    def assert_raster_legend_labels(self, im_expected, expected_labels):
         """Asserts legend correctly describes classified image on Axes ax,
         checking the legend labels and the values
 
         Parameters
         ----------
         im_expected: array of arrays with expected classified image on ax.
-            Class values must start with 0, 1, 2, etc.
-        all_label_options: list of lists
-            Each internal list represents a class and said list is a list of
-            strings where at least one string is expected to be in the legend
-            label for this category. Internal lists must be in the same order
-            as bins in im_expected, e.g. first internal list has the expected
-            label options for class 0.
+        expected_labels: list of lists
+            Each sublist within the expected_labels list contains the word
+            or word variations expected to be found in the legend labels of
+            the plot being tested. Example list: [["gain", "increase"]]
+            would be provided if you wanted to test that the word "gain" OR
+            "increase"  were found in the first legend element.
+            TODO: i think it's or but let's just clarify it's not "and"
+            TODO: we should have tests that check what happens if someone
+            provides only 2 sublist but there are three legend labels.
+            Sublists must be in the same order as the legend elements are
+            in. EXAMPLE: the first sublist will map to the first labeled item
+            in a plot legend.
 
         Returns
         ----------
-        Nothing (if checks pass) or raises error
+        Nothing (if checks pass) or raises assertion error
+        """
 
+        # TODO add test for a plot with no image in it. get_plot_image should
+        # return an error
 
-        Notes
-        ----------
-        First compares all_label_options against the legend labels to find
-        which element of all_label_options matches that entry. E.g. if the
-        first legend entry has a match in the first list in all_label_options,
-        then that legend entry corresponds to the first class (value 0).
-        Then the plot image array is copied and the values are set to the
```

Comment:
@nkorinek this is the part that you removed from the code that i want to add back - checking the data against the legend.